src/main/java/com/zaxxer/hikari/pool/PoolEntry.java

```diff
@@ -90,7 +90,9 @@ void setFutureEol(final ScheduledFuture<?> endOfLife)
 
    Connection createProxyConnection(final ProxyLeakTask leakTask, final long now)
    {
-      return ProxyFactory.getProxyConnection(this, connection, openStatements, leakTask, now, hikariPool.isAutoCommit());
+      ProxyConnection pc = ProxyFactory.getProxyConnection(this, connection, openStatements, leakTask, now);
+      hikariPool.initProxyConnection(pc);
```

Comment:
it would need to duplicate readonly and autocommit in pool entry and touching parameter count check :) I hope jvm would inline these 'simple' methods.